internal/controller/controller.go

```diff
@@ -389,6 +415,10 @@ func (c *Controller) createMeshService(service *corev1.Service) (*corev1.Service
 				targetPort = intstr.FromInt(c.getTCPPortFromState(service.Name, service.Namespace, sp.Port))
 			}
 
+			if targetPort.IntVal == 0 {
+				log.Errorf("Could not get TCP Port for service: %s with service port: %v", service.Name, sp)
```

Comment:
Shoudn't we skip this item in that case using a `continue` statement ?

Reply:
Good catch!